sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala

```diff
@@ -140,6 +141,14 @@ class SQLQueryTestSuite extends QueryTest with SharedSQLContext {
     val input = fileToString(new File(testCase.inputFile))
 
     val (comments, code) = input.split("\n").partition(_.startsWith("--"))
+
+    // Runs all the tests on both codegen-only and interpreter modes. Since explain results differ
+    // when `WHOLESTAGE_CODEGEN_ENABLED` disabled, we don't run these tests now.
+    val codegenConfigSets = Array(("false", "NO_CODEGEN"), ("true", "CODEGEN_ONLY")).map {
+      case (wholeStageCodegenEnabled, codegenFactoryMode) =>
+        Array( // SQLConf.WHOLESTAGE_CODEGEN_ENABLED.key -> wholeStageCodegenEnabled,
```

Comment:
If `wholeStageCodegenEnabled` is not used, let's not complicate the code now.

Reply:
ok